src/java/org/apache/cassandra/cql3/statements/schema/AlterTableStatement.java

```diff
@@ -553,6 +554,33 @@ public void validate(ClientState state)
             validateDefaultTimeToLive(attrs.asNewTableParams());
         }
 
+        private TableParams validateAndUpdateTransactionalMigration(TableParams prev, TableParams next)
+        {
+            boolean modeChange = prev.transactionalMode != next.transactionalMode;
+            boolean wasMigrating = prev.transactionalMigrationFrom != TransactionalMigrationFromMode.none;
+            boolean forceMigrationChange = prev.transactionalMigrationFrom != next.transactionalMigrationFrom;
+
+            if (modeChange && next.transactionalMode.accordIsEnabled && !DatabaseDescriptor.getAccordTransactionsEnabled())
+                throw ire(format("Cannot change transactional mode to %s for %s.%s with accord_transactions_enabled set to false",
+                                 next.transactionalMode, keyspaceName, tableName));
+
+            // user is manually updating migration mode, don't interfere
+            if (forceMigrationChange)
+                return next;
+
+            if (!modeChange)
+                return next;
+
+            if (wasMigrating)
+                throw ire(format("Cannot change transactional mode from %s to %s for %s.%s before transactional migration has completed",
```

Comment:
I think we'd need a migration to occur between the different interop modes as well? Like if we moved from `full` to `migration`, I think we'd want to make sure data written with `full` had been repaired so it would be visible to non-serial reads?

Reply:
You are right some of them have should require a migration, but you should be able to change interop modes.

From Accord's perspective the only mode that matters is `full` (ignores supplied consistency level, does async commit), and not `full`.

For C* the modes you should be able to switch between unsafe <-> mixed <-> migration. Really `migration` is the only mode that you can safely use on anything. The rest make the assumption the application is not going to mix Accord/non-Accord reads/writes on the same data.

Reply:
You are right some of them have should require a migration, but you should be able to change interop modes.

From Accord's perspective the only mode that matters is full (ignores supplied consistency level, does async commit), and not full.

For C* the modes you should be able to switch between unsafe <-> mixed <-> migration. Really migration is the only mode that you can safely use on anything. The rest make the assumption the application is not going to mix Accord/non-Accord reads/writes on the same data.

Reply:
This doesn't prevent you from changing interop modes, but it does prevent you from changing interop mode while there's an unfinished migration. 

Since each of the migration modes provide different consistency guarantees, I think a migration between at least some of them is needed to prevent invalidating programmer assumptions in between them. For instance, you can do normal QUORUM reads during the “migration” (mixed_reads proposed) interop mode. We’ll want to at least run a barrier as a migration so that successfully completed writes don’t become intermittently unreadable since the writes at `full` may have only been persisted on a single node.

When upgrading from an interop mode allowing unsafe writes, don’t think it would be a bad idea to keep the table in a symbolic migrating state for operator visibility and alerting purposes, since the safety of the safer interop mode won’t be available until data is repaired.

As an aside I’d thought the “mixed” interop mode did something like a quorum read for accord operations to prevent non-serial writes from causing multiple outcomes for accord operations, but it looks like I misremembered that (or am not looking hard enough). Assuming I’m not missing something, we should definitely rename mixed to unsafe_writes since non-serial writes can cause multiple outcomes, it’s just a little less dangerous since BRR is routed through accord.

Reply:
OK, I can sort of see that if you are running an unsafe mode in Cassandra it would require a restart of the migration to actually migrate with the level of safety/correctness of the safe mode (`migration`).

I think the pain point is if you set an unsafe mode and want to switch to a safe mode and restart the migration then you can't. You have to complete the unsafe one which requires a repair to complete and you are exposed to data loss the entire time.

Force migration change still works so I think that is good enough for now.

Accord always respects the consistency levels supplied except in `TransactionMode.full` where everything runs through Accord. It's Cassandra that has all the unsafe ways to operate to try and preserve performance assuming that you won't overlap access in unsafe ways.